lib/msal-browser/src/app/IPublicClientApplication.ts

```diff
@@ -12,20 +12,20 @@ import { EndSessionRequest } from "../request/EndSessionRequest";
 import { BrowserConfigurationAuthError } from "../error/BrowserConfigurationAuthError";
 
 export interface IPublicClientApplication {
-    acquireTokenPopup(request: PopupRequest): Promise<AuthenticationResult>;
+    acquireTokenPopup(request: PopupRequest): Promise<AuthenticationResult | null>;
     acquireTokenRedirect(request: RedirectRequest): Promise<void>;
-    acquireTokenSilent(silentRequest: SilentRequest): Promise<AuthenticationResult>;
+    acquireTokenSilent(silentRequest: SilentRequest): Promise<AuthenticationResult | null>;
     addEventCallback(callback: Function): string | null;
     removeEventCallback(callbackId: string): void;
     getAccountByHomeId(homeAccountId: string): AccountInfo | null;
     getAccountByLocalId(localId: string): AccountInfo | null;
     getAccountByUsername(userName: string): AccountInfo | null;
     getAllAccounts(): AccountInfo[];
     handleRedirectPromise(hash?: string): Promise<AuthenticationResult | null>;
-    loginPopup(request?: PopupRequest): Promise<AuthenticationResult>;
+    loginPopup(request?: PopupRequest): Promise<AuthenticationResult | null>;
     loginRedirect(request?: RedirectRequest): Promise<void>;
     logout(logoutRequest?: EndSessionRequest): Promise<void>;
-    ssoSilent(request: SsoSilentRequest): Promise<AuthenticationResult>;
+    ssoSilent(request: SsoSilentRequest): Promise<AuthenticationResult | null>;
```

Comment:
I don't think any of these can actually be null though?

Reply:
Ah, its because some of the underlying functions can return `null`, yeah?

Reply:
Yeah that's right. It looks like there's only one place in one of the functions in common that returns null. If this is something we want to change it would be relatively low impact.

Reply:
I would prefer that we are able to make this change without needing to change the public API, especially give these APIs will never actually return `null` to the user.

Reply:
Done.